### DISCUSSION TITLE
ColdFusion Component for Google Gemini

### DISCUSSION THREAD

**Discussion by giscus** [cfjedimaster/raymondcamden2023#39] [Comments]
# ColdFusion Component for Google Gemini

A note about my open-source ColdFusion component for Google Gemini's GenAI service.

https://www.raymondcamden.com/2024/10/03/coldfusion-component-for-google-gemini

**Comment by charlesr1971** [comment]
Awesome little feature, Ray. I have been playing around with Coldfusion and OpenAi. 
Its pretty interesting stuff. 
The hardest part is making a prompt that returns consistent results. 
I have been building an app that allows devs to practice Angular based technical tests.
I have written many of the questions myself, but I decided I wanted to see if AI could help me, to build these questions. 
Connecting to the API is easy, but getting the model to return a response, with an identical structure is more difficult.

**Reply by cfjedimaster** [reply]
So my CFC doesn't support this, but you can use JSON Schemas to have identical structures returned. See my blog post here, https://www.raymondcamden.com/2024/06/11/using-json-schema-with-google-gemini. And I built a CF demo here - https://github.com/cfjedimaster/gemini-ai-preso/blob/main/demos/coldfusion/json_response3.cfm
